### PR TITLE
Creating a data model

### PR DESCRIPTION
**Pull request by r-ahmed2022**
In this PR, I completed the following:-
- Modified our new rails app i.e. blogsapp
 - Initialized with git
- Setup up Postgres database
- Build project schema including necessary migration files, posts table having a corresponding foreign key to the users table
- All foreign keys columns have a corresponding index.

### CODE REVIEW

**Comment by addod19** [comment]
### Hi @r-ahmed2022, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because you did not follow the Git-flow workflow as seen in the project requirements
Screenshot from 2023-05-03 12-25-31
Thus creating the PR from `branch-model` into `dev` and not `master` as we have currently

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rotimiazeez — COMMENTED** [review]
# Hi @r-ahmed2022 , 

### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that does no have a professional README file 😔. Please ensure all the sections of your README file are filled out according to the project you are doing at the moment 😉.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Comment by r-ahmed2022** [comment]
> # Hi @r-ahmed2022 ,
> ### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> ## Invalid Code Review Request
> You have submitted a project that does no have a professional README file 😔. Please ensure all the sections of your README file are filled out according to the project you are doing at the moment 😉.
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **Does not count** into the code reviews limit._

@rotimiazeez , the readme file is on dev branch , kindly check

File: db/migrate/20230503111751_create_comments.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
```
Comment (danmainah) [inline db/migrate/20230503111751_create_comments.rb:6]: - [ ] @r-ahmed2022 great work on your migration files, however, I would recommend you use author_id when referencing the user in your comments table as it is in the ERD diagram

File: db/migrate/20230503112115_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
```
Comment (danmainah) [inline db/migrate/20230503112115_create_likes.rb:5]: - [ ] @r-ahmed2022 great work on your migration files, however, I would recommend you use author_id when referencing the user in your likes table as it is in the ERD diagram

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @r-ahmed2022 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ All tables were created successfully
✔️ Gitflow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 3]
```diff
@@ -1,24 +1,177 @@
...
+
+## :card_index: Entity Relationship Diagram <a name="er-diagram"></a>
+
+![er-diagram](https://user-images.githubusercontent.com/84629565/203113736-82dacec0-6b75-42e6-b87e-8bf441ffbe57.png)
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- in progress :smile: [Live Demo Link]()
```
[the remark this answers is not in the document]
Comment (r-ahmed2022) [reply README.md:79]: Done, as advised

File: db/migrate/20230503111751_create_comments.rb (hunk 1 above)
Comment (r-ahmed2022) [reply db/migrate/20230503111751_create_comments.rb]: Done, ad advised

File: db/migrate/20230503112115_create_likes.rb (hunk 2 above)
Comment (r-ahmed2022) [reply db/migrate/20230503112115_create_likes.rb]: Done, as advised

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi  @r-ahmed2022, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failing linters. Kindly fix the errors
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rotimiazeez — APPROVED** [review]
Hi @r-ahmed2022 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job  working on all required changes 👏🏾
- All migration file are created as required 
- No Linter errors ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**